### PR TITLE
Suggest naming fix

### PR DESCRIPTION
**Pull request by mahesh-panchal**
## PR checklist

Renames file based on prefix instead of prefixing filename to fix https://nfcore.slack.com/archives/CJRH30T6V/p1669393822497519

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [ ] Remove all TODO statements.
- [ ] Emit the `versions.yml` file.
- [ ] Follow the naming conventions.
- [ ] Follow the parameters requirements.
- [ ] Follow the input/output options guidelines.
- [ ] Add a resource `label`
- [ ] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [ ] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

**Comment by mahesh-panchal** [comment]
Need to fix contains for html files which have date inside.

**Review by drpatelh — APPROVED** [review]
Beautiful 🤩

File: modules/nf-core/fastqc/main.nf [hunk 1]
```diff
@@ -21,9 +21,12 @@ process FASTQC {
...
     def args = task.ext.args ?: ''
     def prefix = task.ext.prefix ?: "${meta.id}"
+    def rename_to = reads instanceof Path || reads.size() == 1 ? "$reads ${prefix}.${reads.extension}" : reads.withIndex().collect { entry, index -> "$entry ${prefix}_${index + 1}.${entry.extension}" }.join(' ')
     """
-    printf "%s\\n" $reads | while read f; do [[ \$f =~ ^${prefix}.* ]] || ln -s \$f ${prefix}_\$f ; done
-    fastqc $args --threads $task.cpus ${prefix}*
+    printf "%s %s\\n" $rename_to | while read old_name new_name; do
+        [ -f "\${new_name}" ] || ln -s \$old_name \$new_name
+    done
+    fastqc $args --threads $task.cpus ${rename_to.tokenize().withIndex().findAll{ entry, index -> index % 2 }.collect{ it[0] }.join(' ')}
```
Comment (drpatelh) [inline modules/nf-core/fastqc/main.nf:29]: Well, this took a turn for the worse! 🤣

Comment (mahesh-panchal) [reply modules/nf-core/fastqc/main.nf:29]: don't I know it... Got a better suggestion.

**Comment by mahesh-panchal** [comment]
Right.. I think that fixed the small things ...

File: modules/nf-core/fastqc/main.nf (hunk 1 above)
Comment (drpatelh) [reply modules/nf-core/fastqc/main.nf]: I don't I'm afraid. I did try playing with it on Friday but it turned into a can of worms when trying to deal with PE reads. This is why I suggested going back to the previous implementation but having another input value for interleaved.

File: modules/nf-core/fastqc/main.nf [hunk 2]
```diff
@@ -21,9 +21,14 @@ process FASTQC {
     script:
     def args = task.ext.args ?: ''
     def prefix = task.ext.prefix ?: "${meta.id}"
+    def old_new_pairs = reads instanceof Path || reads.size() == 1 ? [[ reads, "${prefix}.${reads.extension}" ]] : reads.withIndex().collect { entry, index -> [ entry, "${prefix}_${index + 1}.${entry.extension}" ] }
```
Comment (drpatelh) [inline modules/nf-core/fastqc/main.nf:24]: Looks good to me and kudos to the Groovy magic :bowtie:. But now much more complex to track this logic overall that could be handled by having an additional input value to the process?

Comment (mahesh-panchal) [reply modules/nf-core/fastqc/main.nf:24]: I think going back to the old way introduces much more input and cases, and duplicated code making it less maintainable. There's now tests for multiple fastx and bam too. But then again, if someone trying to update can't read the Groovy magic, that doesn't help either.

**Comment by mahesh-panchal** [comment]
OK to merge in light of all the changes?

**Comment by drpatelh** [comment]
Sure 👍🏽 We can update (revert) later on if need be. Thanks for fixing!

**Comment by drpatelh** [comment]
Feel free to merge when the tests pass

**Pull request merged**